streams/src/main/java/org/apache/kafka/streams/state/internals/StreamThreadStateStoreProvider.java

```diff
@@ -72,8 +82,23 @@ public StreamThreadStateStoreProvider(final StreamThread streamThread) {
         } else {
             throw new InvalidStateStoreException("Cannot get state store " + storeName + " because the stream thread is " +
                                                      state + ", not RUNNING" +
-                                                     (includeStaleStores ? " or REBALANCING" : ""));
+                                                     (storeQueryParams.includeStaleStores() ? " or REBALANCING" : ""));
         }
     }
 
+    private TaskId createKeyTaskId(final String storeName, final Integer partition) {
+        if (partition == null) {
```

Comment:
can we do a null check way at the top when the `StoreQueryParams` is built and avoid this special casing here..

Reply:
Doing this check in stores() leads to an increase in cyclomatic complexity above 16. So, I have added it here.